packages/nightingale-scrollbox/src/nightingale-scrollbox.ts

```diff
@@ -0,0 +1,157 @@
+import NightingaleElement from "@nightingale-elements/nightingale-new-core";
+import { customElement, property } from "lit/decorators.js";
+import { NightingaleScrollboxItem } from "./nightingale-scrollbox-item";
+
+
+@customElement("nightingale-scrollbox")
+export class NightingaleScrollbox<TData> extends NightingaleElement {
+  /** Amount added to the top and bottom side of the bounding box before the intersection test is performed.
+   * Can be a number or CSS length.
+   * This lets you adjust the bounds outward so that an item is considered visible even if it is still hidden but is close to the visible area.
+   * Negative values will adjust the bound inward so that an item is considered hidden even if it is visible but is close to the edge of the visible area. */
+  @property({ type: String })
+  "root-margin"?: string;
+
+  /** If this attribute is set, wheel scrolling will be disabled whenever Ctrl (or Meta/Command) key is pressed.
+   * This helps to prevent bad user experience when some elements in the scrollbox have special Ctrl+Wheel behavior (e.g. zoom) but the gaps between elements still have default scrolling behavior. */
+  @property({ type: Boolean })
+  "disable-scroll-with-ctrl"?: boolean;
+
+  private readonly _items = new Set<NightingaleScrollboxItem<TData>>;
+  /** Set of currently registered items */
+  get items() { return this._items as ReadonlySet<NightingaleScrollboxItem<TData>>; }
+  
+  private observer?: IntersectionObserver;
+
+  private readonly callbacks: {
+    onRegister?: (item: NightingaleScrollboxItem<TData>) => void | Promise<void>,
+    onEnter?: (item: NightingaleScrollboxItem<TData>) => void | Promise<void>,
+    onExit?: (item: NightingaleScrollboxItem<TData>) => void | Promise<void>,
+    onUnregister?: (item: NightingaleScrollboxItem<TData>) => void | Promise<void>,
+  } = {};
+
+  override connectedCallback() {
+    super.connectedCallback();
+    this.initObserver();
+    this.disableScrollWithCtrl(this["disable-scroll-with-ctrl"]);
+  }
+  override disconnectedCallback() {
+    this.dispose();
+    super.disconnectedCallback();
+  }
+  override attributeChangedCallback(name: string, oldValue: string | null, newValue: string | null): void {
+    super.attributeChangedCallback(name, oldValue, newValue);
+    if (this.isConnected && newValue !== oldValue) {
+      if (name === "root-margin") {
+        this.disposeObserver();
+        this.initObserver();
+      }
+      if (name === "disable-scroll-with-ctrl") {
+        this.disableScrollWithCtrl(this["disable-scroll-with-ctrl"]);
+      }
+    }
+  }
+
+  private initObserver() {
+    const margin = normalizeCssLength(this["root-margin"]);
+    this.observer = new IntersectionObserver(entries => this.observerCallback(entries), { root: this, rootMargin: `${margin} 0px ${margin} 0px` });
+    const items = Array.from(this.items);
+    for (const item of items) {
```

Comment:
```suggestion
    for (const item of this.items) {
```
Curious why you needed the `Array.from`?

Reply:
You're right, it's not needed here. I probably copied it from elsewhere. 
(on line 100 it's necessary to have Array.from, otherwise we would be deleting elements from a set while iterating it).